sys/trickle/trickle.c

```diff
@@ -33,25 +35,21 @@ void trickle_callback(trickle_t *trickle)
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t old_interval = trickle->I;
-    uint32_t max_interval = trickle->Imin << trickle->Imax;
-    uint32_t diff = old_interval - trickle->t;
+    assert(trickle->I > 0);
 
-    trickle->I *= 2;
+    uint32_t diff = trickle->I - trickle->t;
 
-    if ((trickle->I == 0) || (trickle->I > max_interval)) {
-        trickle->I = max_interval;
-    }
+    trickle->I *= 2;
+    trickle->I = MIN(trickle->I, (trickle->Imin << trickle->Imax));
 
     DEBUG("trickle: I == %" PRIu32 ", diff == %" PRIu32 "\n", trickle->I, diff);
 
     trickle->c = 0;
-    trickle->t = random_uint32_range(old_interval, trickle->I);
+    trickle->t = random_uint32_range(trickle->I/2, trickle->I);
```

Comment:
not #7163 doesn't fix that, assume the following:

- `trickle->I = 13` and `max_interval = 25`
- hence, `old_interval = 13` and ` trickle->I *= 2 = 26` 
- hence, `trickle-I > max_interval` -> `trickle->I = 25` and `old_interval /= 2 = 6`
- hence, `trickle->t = random_uint32_range(6, 25)` which is wrong, correct would be `trickle->t = random_uint32_range(12,25)`